lib/cooked_post_processor.rb

```diff
@@ -181,10 +183,9 @@ def convert_to_link!(img)
       img.add_class("animated")
     end
 
-    if original_width <= SiteSetting.max_image_width &&
-         original_height <= SiteSetting.max_image_height
-      return
-    end
+    skip_thumbnail =
```

Comment:
How about naming this `generate_thumbnail` so we have all the test be _positive_ (aka. `if generate_thumbnail`) instead of them being _negative_ (aka. `unless skip_thumbnail`)?